### PR TITLE
New package: PostgREST.postgrest version 12.2.3

### LINKED ISSUE #189017: [Package Request]: PostgREST
### How can we help?

I would like someone else to build the manifest.

### Please read and ensure the following

- [x] The installer meets the above requirements

### Please provide the following information

Download Page Url: https://github.com/PostgREST/postgrest/releases/latest
Publisher: PostgREST.org
Package Name: PostgREST
Description: PostgREST is a standalone web server that turns your PostgreSQL database directly into a RESTful API. The structural constraints and permissions in the database determine the API endpoints and operations.
Package Version: 12.2.3
Installer Url: https://github.com/PostgREST/postgrest/releases/download/v12.2.3/postgrest-v12.2.3-windows-x64.zip

### PR DESCRIPTION
**Pull request by Dragon1573**
## Notes

- External dependencies required, add "InstallationNotes" for it.
- See https://github.com/microsoft/winget-pkgs/issues/189017#issuecomment-2460520948

## Checklist for Pull Requests

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Resolve #189017

## Manifests

- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?

Note: `<path>` is the directory's name containing the manifest you're submitting.

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/189441)

### CODE REVIEW

**Comment by stephengillie** [comment]
Automatic Validation ended with:

> Executable C:\Users\validator\AppData\Local\Microsoft\WinGet\Links\postgrest.exe returned exit code: -1073741515

Automated error analysis suggests -1073741515 may mean STATUS_DLL_NOT_FOUND - possibly due to a dependency on `Microsoft.VCRedist.2015+.x64` (or `.x86`).

(Automated response - build 897.)

**Comment by develop7** [comment]
@stephengillie yes, it depends on psql.dll (and its' dependencies)

**Review by ItzLevvie — APPROVED** [review]

**Comment by stephengillie** [comment]
Hi @Dragon1573,

Is a dependency missing? The application refuses to run, instead giving this message on each launch.
image

Verify:
image

**Comment by Dragon1573** [comment]
1. `libpq.dll` can be downloaded and installed via package identifier `PostgreSQL.psqlODBC`.
2. The "dependency" package will not add itself to `$env:PATH`.
3. `postgrest.exe` can't load `libpq.dll` and failed to launch.

@stephengillie , I don't have good solution for resolving this problem. 😞

File: manifests/p/PostgREST/postgrest/12.2.3/PostgREST.postgrest.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+NestedInstallerFiles:
+- RelativeFilePath: artifacts/postgrest-windows-x64/postgrest.exe
+InstallModes:
+- silent
+- silentWithProgress
+UpgradeBehavior: uninstallPrevious
+ReleaseDate: 2024-08-01
+Dependencies:
+  ExternalDependencies:
+  - libpq.dll
```
Comment (Dragon1573) [inline manifests/p/PostgREST/postgrest/12.2.3/PostgREST.postgrest.installer.yaml:17]: `libpq.dll` is mentioned here as an external dependency.

File: manifests/p/PostgREST/postgrest/12.2.3/PostgREST.postgrest.locale.en-US.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+- postgres
+- postgresql
+- postgrest
+- rest
+- server
+- sql
+InstallationNotes: |-
+  "libpq.dll" is required by this package. It can be found in any version of PostgreSQL ODBC drivers.
+  Available via "PostgreSQL.psqlODBC" in WinGet. You should manually add it's "bin/" directory (
+  "$env:PROGRAMFILES/psqlODBC/<version>/bin" by default) into "$env:PATH" environment variable.
```
Comment (Dragon1573) [inline manifests/p/PostgREST/postgrest/12.2.3/PostgREST.postgrest.locale.en-US.yaml:36]: So I note here and asking users to resolve "`libpq.dll` not found" error on their own.

**Review by Dragon1573 — COMMENTED** [review]
Problem is already mentioned before.

- See https://github.com/microsoft/winget-pkgs/issues/189017#issuecomment-2462387942
- See https://github.com/microsoft/winget-pkgs/issues/189017#issuecomment-2462410030

**Review by ItzLevvie — APPROVED** [review]

**Review by ItzLevvie — APPROVED** [review]

**Comment by stephengillie** [comment]
So the package has a dependency, and a workaround is needed to address a bug.

**Review by ItzLevvie — APPROVED** [review]

**Review by ItzLevvie — APPROVED** [review]

**Comment by Dragon1573** [comment]
> [!WARNING]
>
> The package requires `libpq.dll`, which can't be correctly added to `$env:PATH`. Unable to resolve this problem.

**Pull request closed without merging**